binary_search_tree/BinarySearchTree.c

```diff
@@ -0,0 +1,341 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+typedef struct node
+{   // Node for binary search tree
+    int data;
+    struct node * left;
+    struct node * right;
+
+}node;
+
+
+
+
+// Function for insering a node in the tree
+node * insert_node(node * root, int a)
+{
+    if(root == NULL)
+    {
+        root = (node *)malloc(sizeof(node));
+        root->data = a;
+        root->left = NULL;
+        root->right = NULL;
+    }
+    else
+    {
+        node * temp;
+        temp = (node *)malloc(sizeof(node)); // Space allcated for the node
+        temp->data = a;
+        temp->left = NULL;
+        temp->right = NULL;
+        node * temp1 = root;
+        while(1)
+        {
+            int key = temp1->data;
+            if(key<=a){
+                if(temp1->right==NULL){
+                    temp1->right = temp;
+                    break;
+                }
+                else
+                {
+                    temp1=temp1->right;
+                }
+            }
+            else{
+                if(temp1->left==NULL){
+                    temp1->left = temp;
+                    break;
+                }
+                else
+                {
+                    temp1=temp1->left;
+                }
+
+            }
+
+        }
+
+
+
+    }
+
+    return root;
+}
+
+// Function for swapping node values
+void swap(node *a, node *b){
+    int temp1 = a->data;
+    a->data = b->data;
+    b->data = temp1;
+}
+
+// Function for finding the smallest node in the right subtree.
+node * successor(node * temp, node * temp1)
+{
```

Comment:
Consistent bracket order. Either in the same line or in the new line in whole code